.ibm/pipelines/lib/README.md

```diff
@@ -0,0 +1,88 @@
+# Pipeline Library Modules
+
+Utility functions organized by responsibility. See `../README.md` for development guidelines.
+
+## Modules
+
+### `log.sh`
+
+Structured logging with ANSI colors.
+
+Functions: `log::info`, `log::warn`, `log::error`, `log::success`, `log::debug`, `log::hr`
+
+### `common.sh`
+
+Common utilities.
+
+Functions: `common::oc_login`, `common::is_openshift`, `common::sed_inplace`,
+`common::get_previous_release_version`
+
+### `k8s-wait.sh`
+
+Kubernetes wait/polling operations.
+
+Functions: `k8s_wait::deployment`, `k8s_wait::job`, `k8s_wait::service`, `k8s_wait::endpoint`,
+`k8s_wait::backstage_resource`
+
+### `operators.sh`
+
+Operator and OLM installations.
+
+Functions: `operator::install_subscription`, `operator::check_status`,
+`operator::install_postgres_ocp`, `operator::install_postgres_k8s`,
+`operator::install_serverless_logic`, `operator::install_serverless`, `operator::install_pipelines`,
+`operator::install_tekton`, `operator::delete_tekton`, `operator::install_olm`,
+`operator::uninstall_olm`
+
+## Creating New Modules
+
+### Structure
+
+```bash
+#!/usr/bin/env bash
+
+# Module: <module_name>
+# Description: <brief description>
+# Dependencies: <tools required>
+
+set -euo pipefail
```

Comment:
This `set` will be carried over to the whole script and affect the whole execution. 
I'd keep this set up only on the entry point (openshift-ci-tests.sh at the moment) and refrain from redefining it **or** the `set` in the library should be aligned with the `set` in the entrypoint.